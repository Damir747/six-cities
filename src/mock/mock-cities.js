const cities = {
  'Paris': { lat: 48.85661, lng: 2.351499, zoom: 13 },
  'Cologne': { lat: 50.938361, lng: 6.959974, zoom: 13 },
  'Brussels': { lat: 50.846557, lng: 4.351697, zoom: 13 },
  'Amsterdam': { lat: 52.37454, lng: 4.897976, zoom: 13 },
  'Hamburg': { lat: 53.550341, lng: 10.000654, zoom: 13 },
  'Dusseldorf': { lat: 51.225402, lng: 6.776314, zoom: 13 },
};

export default cities;
