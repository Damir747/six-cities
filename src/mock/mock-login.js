const loginName = 'Oliver.conner@gmail.com';
export default loginName;
