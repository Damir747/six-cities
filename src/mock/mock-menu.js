const menuUpArray = [
  {
    id: 0,
    title: 'Popular',
  },
  {
    id: 1,
    title: 'Price: low to high',
  },
  {
    id: 2,
    title: 'Price: high to low',
  },
  {
    id: 3,
    title: 'Top rated first',
  }];

export { menuUpArray };
