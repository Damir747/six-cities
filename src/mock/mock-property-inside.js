const propertyInside = [
  {
    id: 0,
    title: 'Wi-Fi'
  },
  {
    id: 1,
    title: 'Washing machine'
  },
  {
    id: 2,
    title: 'Towels'
  },
  {
    id: 3,
    title: 'Heating'
  },
  {
    id: 4,
    title: 'Coffee machine'
  },
  {
    id: 5,
    title: 'Baby seat'
  },
  {
    id: 6,
    title: 'Kitchen'
  },
  {
    id: 7,
    title: 'Dishwasher'
  },
  {
    id: 8,
    title: 'Cabel TV'
  },
  {
    id: 9,
    title: 'Fridge'
  }
];

export default propertyInside;
