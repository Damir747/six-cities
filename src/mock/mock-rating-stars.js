const stars = [
  {
    id: 5,
    value: 5,
    title: 'perfect'
  },
  {
    id: 4,
    value: 4,
    title: 'good'
  },
  {
    id: 3,
    value: 3,
    title: 'not bad'
  },
  {
    id: 2,
    value: 2,
    title: 'badly'
  },
  {
    id: 1,
    value: 1,
    title: 'terribly'
  },
];

export default stars;
