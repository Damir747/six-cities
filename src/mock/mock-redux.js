import { redirect } from "../store/middleware/redirect";

const mockRedux = () => {
  const store = {
    getState: jest.fn(() => ({})),
    dispatch: jest.fn(),
  };
  const next = jest.fn();
  const invoke = (action) => redirect(action)(next)(action);
  return { store, next, invoke };
};

export { mockRedux };
