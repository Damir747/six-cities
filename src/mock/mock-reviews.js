const reviews = [{
  id: 0,
  author: 'Max',
  img: 'img/avatar-max.jpg',
  reviewText: 'A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam. The building is green and from 18th century.',
  rating: 80,
  date: 'April 2019'
}];

export default reviews;
