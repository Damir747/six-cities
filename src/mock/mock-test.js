import cities from '../mock/mock-cities';
import propertyInside from '../mock/mock-property-inside';
import { AuthorizationStatus } from '../const';

const initialMockState = {
  HOTEL: {
    hotel: null,
    rooms: [{
      bookmark: "To bookmarks",
      card: "Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.",
      city: { name: "Brussels", location: { latitude: 50.846557, longitude: 4.351697, zoom: 13 } },
      cityName: "Brussels",
      coordinates: { lat: 50.827557, lng: 4.336697, zoom: 16 },
      description: "Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.",
      features: [],
      goods: ["Laptop friendly workspace", "Breakfast"],
      host: { title: "Meet the host", user: {} },
      id: 1,
      images: [],
      img: "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg",
      level: "",
      priceText: "night",
      priceValue: "939",
      rating: 76,
      title: "Loft Studio in the Central Area",
      type: "house",
    }],
    neighbourhood: [{
      city: {},
      images: [],
      title: '',
      rating: '',
      type: '',
      goods: [],
      host: {},
      description: '',
      id: 2,
      level: '',
      img: '',
      priceValue: '',
      priceText: '',
      bookmark: 'To bookmarks',
      card: '',
    }],
    isHotelListLoading: false,
    isHotelListLoaded: false,
    reviews: [],
    isHotelLoading: false,
    isHotelLoaded: false,
    isCommentLoading: false,
    isCommentLoaded: false,
    isNeighbourhoodLoading: false,
    isNeighbourhoodLoaded: false,
  },
  CITY: {
    activeCity: 'Brussels',
    cities,
    currentCity: 'Brussels',
    isCityListLoaded: false,
    isCityListLoading: false,
  },
  FAVORITES: {
    favorites: [{
      bookmark: "To bookmarks",
      card: "Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.",
      city: { name: "Brussels", location: { latitude: 50.846557, longitude: 4.351697, zoom: 13 } },
      cityName: "Brussels",
      coordinates: { lat: 50.827557, lng: 4.336697, zoom: 16 },
      description: "Peaceful studio in the most wanted area in town. Quiet house Near of everything. Completely renovated. Lovely neighbourhood, lot of trendy shops, restaurants and bars in a walking distance.",
      features: [],
      goods: ["Laptop friendly workspace", "Breakfast"],
      host: { title: "Meet the host", user: {} },
      id: 1,
      images: [],
      img: "https://assets.htmlacademy.ru/intensives/javascript-3/hotel/18.jpg",
      level: "",
      priceText: "night",
      priceValue: "939",
      rating: 76,
      title: "Loft Studio in the Central Area",
      type: "house",
    }],
    isFavoriteListLoading: false,
    isFavoriteListLoaded: false,
  },
  COMMENT: {
    comment: '',
  },
  INIT: {
    propertyInside,
  },
  LOGIN: {
    loginName: '',
    authorizationStatus: AuthorizationStatus.NO_AUTH,
  },
  NOTIFICATION: {
    notifications: [],
  },
  SORT: {
    sort: 0,
  }
};

export { initialMockState };
