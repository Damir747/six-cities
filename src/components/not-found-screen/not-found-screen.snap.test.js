import React from 'react';
import { render } from '@testing-library/react';
import { Router } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { createMemoryHistory } from 'history';
import * as redux from 'react-redux';
import { initialMockState } from '../../mock/mock-test';
import NotFoundScreen from './not-found-screen';

it(`Should NotFoundScreen render correctly`, () => {
  const mockStore = configureStore({});
  const history = createMemoryHistory();
  const store = mockStore(initialMockState);
  store.dispatch = () => Promise.resolve();

  const { container } = render(
    <redux.Provider store={store}>
      < Router history={history}>
        <NotFoundScreen />
      </ Router>
    </redux.Provider>
  );
  expect(container).toMatchSnapshot();
});
