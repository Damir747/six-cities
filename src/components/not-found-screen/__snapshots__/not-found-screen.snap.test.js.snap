// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should NotFoundScreen render correctly 1`] = `
<div>
  <div
    style="display: none;"
  >
    <svg
      xmlns="http://www.w3.org/2000/svg"
    >
      <symbol
        data-testid="icon-arrow-select"
        id="icon-arrow-select"
        viewBox="0 0 7 4"
      >
        <path
          clip-rule="evenodd"
          d="M0 0l3.5 2.813L7 0v1.084L3.5 4 0 1.084V0z"
          fill-rule="evenodd"
        />
      </symbol>
      <symbol
        data-testid="icon-bookmark"
        id="icon-bookmark"
        viewBox="0 0 17 18"
      >
        <path
          d="M3.993 2.185l.017-.092V2c0-.554.449-1 .99-1h10c.522 0 .957.41.997.923l-2.736 14.59-4.814-2.407-.39-.195-.408.153L1.31 16.44 3.993 2.185z"
        />
      </symbol>
      <symbol
        data-testid="icon-star"
        id="icon-star"
        viewBox="0 0 13 12"
      >
        <path
          clip-rule="evenodd"
          d="M6.5 9.644L10.517 12 9.451 7.56 13 4.573l-4.674-.386L6.5 0 4.673 4.187 0 4.573 3.549 7.56 2.483 12 6.5 9.644z"
          fill-rule="evenodd"
        />
      </symbol>
    </svg>
  </div>
  <div
    class="page"
  >
    <header
      class="header"
    >
      <div
        class="container"
      >
        <div
          class="header__wrapper"
        >
          <div
            class="header__left"
          >
            <a
              class="header__logo-link header__logo-link--active"
              href="/"
            >
              <img
                alt="6 cities logo"
                class="header__logo"
                height="41"
                src="img/logo.svg"
                width="81"
              />
            </a>
          </div>
          <button
            hidden=""
            style="height: 50%; width: 80px; margin: 10px;"
          >
            Logout
          </button>
          <nav
            class="header__nav"
          >
            <ul
              class="header__nav-list"
            >
              <li
                class="header__nav-item user"
              >
                <a
                  class="header__nav-link header__nav-link--profile"
                  href="/login"
                >
                  <div
                    class="header__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      src=""
                    />
                  </div>
                  <span
                    class="header__user-name user__name"
                  >
                    Sign in
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </header>
    <h1>
      404 Not Found
    </h1>
    <a
      href="/"
    >
      Нажмите, чтобы перейти на главную страницу.
    </a>
    <footer
      class="footer container"
    >
      <a
        class="footer__logo-link"
        href="/"
      >
        <img
          alt="6 cities logo"
          class="footer__logo"
          data-testid="footer__logo"
          height="33"
          src="img/logo.svg"
          width="64"
        />
      </a>
    </footer>
  </div>
</div>
`;
